source/agora/flash/Node.d

```diff
@@ -1418,8 +1419,15 @@ public class FlashNode : FlashControlAPI
             return FeeUTXOs.init;
         // Always pay with the node key
         auto utxos = this.listener.getFeeUTXOs(this.conf.key_pair.address, per_byte);
+
+        // calculate a expected fee for checking the fee is correct
+        per_byte = this.listener.getEstimatedTxFee();
+        utxos.utxos.each!(hash =>
+            tx_size += Input(hash, genKeyUnlock(SigPair.init)).sizeInBytes());
+        per_byte.mul(tx_size);
+
         // reuse total_value as refund amount
-        if (!utxos.total_value.sub(per_byte))
+        if (!utxos.total_value.sub(utxos.total_fee) || per_byte != utxos.total_fee)
```

Comment:
If we are gonna add a new `total_fee` field, don't subtract that from the `total_value`. So that the caller knows how much the UTXO set it received is worth and how much fees it should pay with them. Caller should do the `total_value - total_fee` to properly create the refund output.

Reply:
I agree. So I moved the code to the caller and removed the check code for the expected fee because it's not so useful.